### PR TITLE
Implement rearranging of file tabs

### PR DESCRIPTION
**Pull request by katjanakosic**
In this PR, I included the functionality for rearranging file tabs in the code editor. Since we implemented the file tabs using a ForEach loop and ZStacks, it wasn't easy to implement this functionality from scratch. As a solution, I used an already existing package for [reorderable ForEach](https://github.com/globulus/swiftui-reorderable-foreach) loops.

https://user-images.githubusercontent.com/64956154/213113310-e9a40a97-5c0f-4f1d-a980-f1f2d95e97f5.mov

### CODE REVIEW

File: Themis/Views/Assessment/CodeEditor/TabsView.swift [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
 
     var body: some View {
         ScrollView(.horizontal, showsIndicators: false) {
             ScrollViewReader { scrollReader in
                 HStack(spacing: 0) {
                     Divider()
-                    ForEach(cvm.openFiles, id: \.path) { file in
+                    
+                    // ForEach(cvm.openFiles, id: \.path) { file in
+                    ReorderableForEach($cvm.openFiles, allowReordering: $allowReordering) { file, _ in
```
Comment (tomrudnick) [inline Themis/Views/Assessment/CodeEditor/TabsView.swift]:
```suggestion
                    ReorderableForEach($cvm.openFiles, allowReordering: .constant(true)) { file, _ in
```

File: Themis/Views/Assessment/CodeEditor/TabsView.swift [hunk 2]
```diff
@@ -1,15 +1,20 @@
 import SwiftUI
+import SwiftUIReorderableForEach
+
 
 // used to show opened tabs on top of CodeView
 struct TabsView: View {
     @ObservedObject var cvm: CodeEditorViewModel
+    @State private var allowReordering = true
```
Comment (tomrudnick) [inline Themis/Views/Assessment/CodeEditor/TabsView.swift]:
```suggestion
```

**Review by tomrudnick — COMMENTED** [review]
I think this allowReordering should be a constant because the @State is never used.
But it works very nicely!

**Review by botrufus — APPROVED** [review]

File: Themis/Views/Assessment/CodeEditor/TabsView.swift (hunk 1 above)
Comment (katjanakosic) [reply Themis/Views/Assessment/CodeEditor/TabsView.swift]: You're right. I just pushed the changes 👍🏻

**Review by tomrudnick — APPROVED** [review]

**Review by huber-florian — APPROVED** [review]
Good job!

**Pull request merged**